rwengine/src/engine/GameWorld.cpp

```diff
@@ -678,6 +678,26 @@ int GameWorld::getMinute()
 	return state->basic.gameMinute;
 }
 
+void GameWorld::offsetGameTime(int minutes)
+{
+	int gameMinute = state->basic.gameMinute;
+	int gameHour   = state->basic.gameHour;
+
+	gameMinute += minutes;
+	bool backwards = gameMinute < 0;
+
+	gameHour   += gameMinute / 60 - (backwards ? 1 : 0);
+
+	// Black mathgic? No. If the value is negative we want to wrap it around
+	// to the other side, e.g. minute -7 should be 53.
+	// The equivalent would be doing "while (gameMinute < 0) { gameMinute += 60 }"
```

Comment:
I felt the comment should have made clearer why 2 modulos are necessary:

* Input: 7
* Result of first modulo: -7
* Result of addition: 53
* Result of second modulo: 53

different example:

* Input: 0
* Result of first modulo: 0
* Result of addition: 60
* Result of second modulo: 0